transport.go

```diff
@@ -48,33 +53,31 @@ func encodeGetAllWebhooksResponse(ctx context.Context, rw http.ResponseWriter, r
 	return err
 }
 
-func decodeAddWebhookRequest(ctx context.Context, r *http.Request) (interface{}, error) {
-	requestPayload, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		return nil, err
-	}
-	var webhook Webhook
+func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc {
+	return func(c context.Context, r *http.Request) (request interface{}, err error) {
+		requestPayload, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return nil, err
+		}
+		var webhook Webhook
 
-	err = json.Unmarshal(requestPayload, &webhook)
-	if err != nil {
-		// TODO: This is not part of our swagger but I decided to keep it as it was part of the
-		// codebase: https://github.com/xmidt-org/webpa-common/blob/7740b009eb2cada45954289240d73626e82ccb0d/webhook/webhook.go#L76
-		// We could add a counter to see if this is hit in production and decide to remove it based on that.
-		webhook, err = getFirstFromList(requestPayload)
+		err = json.Unmarshal(requestPayload, &webhook)
+		if err != nil {
+			webhook, err = getFirstFromList(config.webhookLegacyDecodeCount, requestPayload)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
If you didn't want to pass the metric to the function, you could increment it after this line.

Reply:
Good idea, I'll add that 🚀